src/org.mate.marco.gschema.xml

```diff
@@ -202,6 +202,18 @@
         <summary>Icon size</summary>
         <description>Size of the application icons displayed in alt-tab popup window. The screen's scale factor is applied to this value.</description>
     </key>
+    <key name="alt-tab-max-columns" type="i">
+        <range min="1" max="1024"/>
+        <default>5</default>
+        <summary>Maximum number of columns in alt-tab popup window</summary>
+        <description>The popup window may be expanded by empty columns up to the maximum columns width to fit window labels if less columns are displayed and if config option 'alt-tab-max-empty-columns' has a value > 0.</description>
+    </key>
+    <key name="alt-tab-max-empty-columns" type="i">
```

Comment:
Would it make sense to change this to `alt-tab-expand-to-fit-title` or something like that and make it a `boolean`? The concept of adding arbitrary columns for padding seems a bit kludgy. In `false` mode, it would be the current behavior (expand only to fit icons), in `true` mode expand the alt-tab window to fit window titles.

Reply:
With a boolean value we will loose the possibility to add _some_ space for long labels but not up to the maximum according to max-columns. Instead of columns one could specify extra space in abstract pixels or percentage relative to max popup width of how much space could be added for long labels.

Reply:
hmm... not sure I fully understand. The Alt+Tab window should expand to either fit the icons or to fit the longest label, not sure why any intermediate step is necessary. Adding arbitrary space/pixels/padding seems like a theme issue, not a user setting.

Reply:
It's not about extra arbitrary space, it's about how much space is added for longer labels. The amount of added space for longer labels is not only limited by the label length, it's also limited by the max-columns parameter (and perhaps also by max-empty-columns).

Perhaps it may be that the user wants to limit the amount of empty space added around icons if larger labels are present. You yourself were first irritated by the large amount of space added, this space could be limited by the max-empty-column parameter. Did you see my screenshots above demonstrating how extra space my be limited using the max-empty-column parameter? If we have only a single flag this flexibility will be lost. Anyway I can also live with a flag.

Reply:
Ok, I changed it as you suggested, see commit  9f66e47 below. We will loose some flexibility in configuring alt-tab layout but perhaps it is now easier to understand.